website/docs/language/style.mdx

```diff
@@ -0,0 +1,729 @@
+├── database
+│   ├── main.tf
+│   ├── outputs.tf
+│   └── variables.tf
+└── networking
+    ├── main.tf
+    ├── outputs.tf
+    └── variables.tf
+```
+
+</CodeBlockConfig>
+
+In this scenario, you would create three workspaces per environment. For example, your production environment would have a "prod-compute", "prod-database", and "prod-networking" workspace. Read more about [Terraform workspace and project best practices](/well-architected-framework/operational-excellence/operational-excellence-workspaces-projects). 
+
+If you do not use Terraform Cloud or Terraform Enterprise, we recommend that you use modules to encapsulate your configuration, and use a directory for each environment. The configuration in each of these directories would call the local modules, each with parameters specific to their environment. This also lets you maintain separate variable and backend configurations for each environment. 
+
+<CodeBlockConfig hideClipboard>
+
+```
+├── modules
+│   ├── compute
+│   │   └── main.tf
+│   ├── database
+│   │   └── main.tf
+│   └── network
+│       └── main.tf
+├── dev
+│   ├── backend.tf
+│   ├── main.tf
+│   └── variables.tf
+├── prod
+│   ├── backend.tf
+│   ├── main.tf
+│   └── variables.tf
+└── staging
+    ├── backend.tf
+    ├── main.tf
+    └── variables.tf
+```
+
+</CodeBlockConfig>
+
+## State sharing
+
+Since your state contains sensitive information, avoid sharing full state files when possible. 
+
+If you use Terraform Cloud or Terraform Enterprise and a resource is managed in a separate workspace, use the [`tfe_outputs`](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/data-sources/outputs) data source to access outputs across workspaces.
+
+If you do not use Terraform Cloud or Terraform Enterprise but still need to reference data about other infrastructure resources, use data sources to query the provider. For example, you can use the [`aws_instance` data source](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/instance) to look up an AWS EC2 instance by its ID or tags.
+
+If neither of these options are viable, you can use the [`terraform_remote_state` data source](/terraform/language/state/remote-state-data). If you use this data source with Terraform Cloud or Terraform Enterprise, configure which workspaces have access to your state using the principle of least privilege. Refer to [remote state sharing](/terraform/cloud-docs/workspaces/settings#remote-state-sharing) for more information.
+
+## Secrets management
+
+If you do not configure remote state storage, the Terraform CLI stores the entire state in plaintext on the local disk. State can include sensitive data, such as passwords and private keys. 
+
+Terraform Cloud and Terraform Enterprise provide state encryption through HashiCorp Vault. 
+
+- When using Terraform Enterprise, we recommend that you define and enforce a Sentinel policy to prevent use of the `local_exec` provisioner or external data sources.
```

Comment:
```suggestion
- When using Terraform Enterprise, define and enforce a Sentinel policy to prevent use of the `local_exec` provisioner or external data sources.
```